src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/bolt/FlowCacheBolt.java

```diff
@@ -88,8 +87,17 @@ protected void init() {
     @Override
     protected void handleInput(Tuple input) throws PipelineException {
         if (active) {
-            if (ComponentId.TICK_BOLT.name().equals(input.getSourceComponent())) {
-                flowCacheService.processFlowLatencyCheck();
+            if (ComponentId.FLOW_STATE_CACHE_BOLT.name().equals(input.getSourceComponent())) {
+                if (FLOW_UPDATE_STREAM_ID.name().equals(input.getSourceStreamId())) {
+                    InfoData payload = pullValue(input, INFO_DATA_FIELD, InfoData.class);
+                    UpdateFlowInfo updateFlowInfo = (UpdateFlowInfo) payload;
```

Comment:
In case of type unsafe transport, better check what exactly we have received.

Reply:
Fixed.